framework/src/dm/arch/sidk_s5jt200/s5j_dm_connectivity.c

```diff
@@ -385,3 +397,201 @@ int dm_conn_unregister_linkdown_cb(conn_cb cb)
+	} while (iterator != NULL);
+
+	curr_record->next = NULL;
+
+error_with_record:
+
+	return DM_ERROR_NONE;
+}
+
+int dm_conn_free_scan_result(dm_scan_info_t **result)
+{
+	dm_scan_info_t *curr_record;
+	dm_scan_info_t *next_record;
+
+	curr_record = *result;
+	while (curr_record != NULL) {
+		next_record = curr_record->next;
+		free(curr_record);
+		curr_record = next_record;
+	}
+
+	*result = NULL;
+	return DM_ERROR_NONE;
+}
+
+
+int dm_conn_wifi_scan(void)
+{
+	int8_t scan_ret;
+	scan_ret = WiFiScanNetwork();
+	if (scan_ret == SLSI_STATUS_SUCCESS) {
+		WiFiFreeScanResults(&wifiscan_result);
+		WiFiGetScanResults(&wifiscan_result);
+		WiFiNetworkJoin((uint8_t *) CONFIG_DM_AP_SSID, strlen(CONFIG_DM_AP_SSID), \
+						NULL, (slsi_security_config_t *) getSecurityConfig(CONFIG_DM_AP_SECURITY, \
+								CONFIG_DM_AP_PASS, SLSI_WIFI_STATION_IF));
+		return DM_ERROR_NONE;
+	}
+	return DM_ERROR_UNKNOWN;
+}
+
+int dm_conn_wifi_connect(conn_cb linkUpEvent, conn_cb linkDownEvent)
+{
+	uint8_t ret;
+	uint8_t result;
+	int8_t ret_isConnected;
+
+	if (g_dm_connected == 1) {
+		dmdbg("Already, Init!\n");
+		return DM_ERROR_NONE;
+	}
+
+	dm_conn_register_linkup_cb(linkUpEvent);
+	dm_conn_register_linkdown_cb(linkDownEvent);
+
+	g_dm_connected = 0;
+	ret_isConnected = WiFiIsConnected(&result, NULL);
+	if (ret_isConnected == SLSI_STATUS_SUCCESS) {
+		dmdbg("Wi-Fi status - Already Connected : %d\n", result);
+		g_dm_connected = 1;
+		return 1;
+	} else if (ret_isConnected != SLSI_STATUS_NOT_STARTED)
+		return DM_ERROR_UNKNOWN;
+
+
+	ret = WiFiStart(SLSI_WIFI_STATION_IF, NULL);
+	if (ret == SLSI_STATUS_SUCCESS) {
+		dmdbg("STA mode started\n");
+		ret = WiFiNetworkJoin((uint8_t *)CONFIG_DM_AP_SSID, strlen(CONFIG_DM_AP_SSID), \
+							  NULL, (slsi_security_config_t *) getSecurityConfig(CONFIG_DM_AP_SECURITY, CONFIG_DM_AP_PASS, SLSI_WIFI_STATION_IF));
+		sleep(1);
+		if (ret == SLSI_STATUS_SUCCESS) {
+			dmdbg("Start doJoin with SSID %s\n", CONFIG_DM_AP_SSID);
+			return DM_ERROR_NONE;
+		}
+		return DM_ERROR_UNKNOWN;
+	}
+	return DM_ERROR_UNKNOWN;
+}
+
+int dm_conn_wifi_disconnect(void)
+{
+	g_dm_connected = 0;
+	return DM_ERROR_NONE;
+}
+
+static int prv_dm_conn_dhcp(void)
+{
+	struct dhcpc_state state;
+	void *dhcp_handle;
+	int ret;
+
+	dhcp_handle = dhcpc_open(DM_NET_DEVNAME);
+	ret = dhcpc_request(dhcp_handle, &state);
+	if (ret != OK) {
+		dhcpc_close(dhcp_handle);
+		return -1;
+	}
+
+	netlib_set_ipv4addr(DM_NET_DEVNAME, &state.ipaddr);
+	netlib_set_ipv4netmask(DM_NET_DEVNAME, &state.netmask);
+	netlib_set_dripv4addr(DM_NET_DEVNAME, &state.default_router);
+
+	dmdbg("IP address : %s ----\n", inet_ntoa(state.ipaddr));
+	return 1;
+}
+
+int dm_conn_dhcp_init(void)
+{
+	uint8_t result;
+	int8_t ret_isConnected;
+
+	if (g_dm_connected == -1) {
+		dmdbg("First, Do Conn Init!\n");
+		return DM_ERROR_UNKNOWN;
+	}
+
+	if (g_dm_connected == 0) {
+		int waitTime = 10;
+		while (waitTime--) {
+			ret_isConnected = WiFiIsConnected(&result, NULL);
```

Comment:
'result' is hard to understand which value asigned when it success or failed.
please rename like a connected_num instead of result (Assuming result means number of available connection, if not you can rename to another thing)